m4/check_pkg_nvtx.m4

```diff
@@ -0,0 +1,72 @@
+# -*- autoconf -*-
+#
+# Copyright (c) 2024      Amazon.com, Inc. or its affiliates. All rights reserved.
+#
+# See LICENSE.txt for license information
+#
+
+AC_DEFUN([CHECK_PKG_NVTX], [
+  check_pkg_found=yes
+
+  check_pkg_CPPFLAGS_save="${CPPFLAGS}"
+  check_pkg_LDFLAGS_save="${LDFLAGS}"
+  check_pkg_LIBS_save="${LIBS}"
+
+  AC_ARG_WITH([nvtx],
+     [AS_HELP_STRING([--with-nvtx=DIR], [Enable tracing capability with NVTX @<:@default=no@:>@])])
+
+  AS_IF([test "${with_nvtx}" = "yes"],
+        [],
+        [test "${with_nvtx}" = "no"],
+        [check_pkg_found=no],
+        [AS_IF([test -d ${with_nvtx}/lib64], [check_pkg_libdir="lib64"], [check_pkg_libdir="lib"])
+         CPPFLAGS="-I${with_nvtx}/include ${CPPFLAGS}"
+         LDFLAGS="-L${with_nvtx}/${check_pkg_libdir} ${LDFLAGS}"])
+
+  AS_IF([test "${check_pkg_found}" = "yes"],
+        [AC_CHECK_LIB([nvToolsExt], [nvtxRangePop], [], [check_pkg_found=no])])
+
+  AS_IF([test "${check_pkg_found}" = "yes"],
+        [check_pkg_define=1
+         $1],
+        [check_pkg_define=0
+         CPPFLAGS="${check_pkg_CPPFLAGS_save}"
+         LDFLAGS="${check_pkg_LDFLAGS_save}"
+         LIBS="${check_pkg_LIBS_save}"
+         $2])
+
+  AC_DEFINE_UNQUOTED([HAVE_NVTX_TRACING], [${check_pkg_define}], [Defined to 1 if NVTX is available])
+
+  AS_IF([test "${check_pkg_found}" = "yes"],
+        [n_nvtx_domain_per_comm=8
+         nvtx_trace_per_comm=1
+         nvtx_trace_per_dev=0
+
+         AC_ARG_ENABLE([n_nvtx_domain_per_comm],
+                       AS_HELP_STRING([--enable-n-nvtx-domain-per-comm=N], [Number of NVTX domains to create per-communicator.
+                       For best usability, this should be equal to the max number of inflights per communicator. Default: 8]),
+                       [n_nvtx_domain_per_comm=${enableval}],
+                       [])
+         AC_ARG_ENABLE([nvtx_trace_per_comm],
+                       AS_HELP_STRING([--enable-nvtx-trace-per-comm], [Collect NVTX traces in a "per-communicator" view, which associates parent
+                        send/recv events with constituent events (segments, controls).]),
+                       AS_IF([test "${enableval}" = "yes"], [nvtx_trace_per_comm=1], [nvtx_trace_per_comm=0]),
+                       [])
+         AC_ARG_ENABLE([nvtx_trace_per_dev],
+                       AS_HELP_STRING([--enable-nvtx-trace-per-dev], [Collect NVTX traces in a "per-device" view, which associates sub-events with
+                        an EFA device, showing activity on each device.]),
+                       AS_IF([test "${enableval}" = "yes"], [nvtx_trace_per_dev=1], [nvtx_trace_per_dev=0]),
+                       [])
```

Comment:
Don't these have to be mutually exclusive, and so this can be a single --flag={comm/dev}?

Reply:
Yeah... as I mentioned before in this review, I had wanted to make enabling both of them possible. But reading over the code again, I don't think that will work the way it is currently written, so yes, I will make them mutually exclusive for now.

Reply:
Done, but I want to keep them as two separate options, because after doing a little extra work we can support both simultaneously, which may be helpful in some cases.